src/diffusers/utils/state_dict_utils.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+State dict utilities: utility methods for converting state dicts easily
+"""
+import enum
+
+
+class StateDictType(enum.Enum):
+    """
+    The mode to use when converting state dicts.
+    """
+
+    DIFFUSERS_OLD = "diffusers_old"
+    # KOHYA_SS = "kohya_ss" # TODO: implement this
```

Comment:
No need for this as `LoraLoaderMixin.lora_state_dict` does that which is input to the `LoraLoaderMixin.load_lora_into_text_encoder`

Reply:
Yeah there's a pseudo-private method in the `LoraLoaderMixin` class that converts a Kohya checkpoint (after detection) to `diffusers` format:

https://github.com/huggingface/diffusers/blob/24563ca654f6574dae93aeece8eeef69e39097e5/src/diffusers/loaders.py#L1841